<launch>
    <!-- Common Interface -->
    <arg name="device" default="192.168.1.20"/>
    <arg name="rate" default="10"/>

    <!-- Camera Settings -->
    <arg name="ip_address" default="$(arg device)"/>
    <arg name="camera_name" default="flir_ax5"/>
    <arg name="camera" default="thermal"/>
    <arg name="frame_id" default="$(arg camera)"/>
    <arg name="calib_url" default=""/>
    <arg name="fps" default="$(arg rate)"/>
    <arg name="raw" default="true"/>

    <!-- Node Settings -->
    <arg name="output" default="screen"/>
    <arg name="view" default="false"/>
    <arg name="calib" default="false"/>
    <arg name="calib_proc" default="false"/>
    <arg name="thermal_proc" default="true"/>

    <!-- Node -->
    <node pkg="flir_gige" type="flir_gige_node" name="$(arg camera)"
        output="$(arg output)">
        <param name="identifier" type="string" value="$(arg ip_address)"/>
        <param name="camera_name" type="string" value="$(arg camera_name)"/>
        <param name="calib_url" type="string" value="$(arg calib_url)"/>
        <param name="frame_id" type="string" value="$(arg frame_id)"/>
        <param name="fps" type="double" value="$(arg fps)"/>
        <param name="raw" type="bool" value="$(arg raw)"/>
    </node>

    <!-- View -->
    <arg name="image" default="image_raw"/>
    <node if="$(arg view)"
        pkg="image_view" type="image_view" name="viewer"
        output="$(arg output)" ns="$(arg camera)" args="image:=$(arg image)">
    </node>

    <!--Calib proc-->
    <node if="$(arg calib_proc)"
        pkg="flir_gige" type="calib_proc_node" name="calib_proc"
        ns="$(arg camera)">
    </node>

    <!-- Calib -->
    <group if="$(arg calib)">
        <arg name="pattern" default="chessboard"/>
        <arg name="size"/>
        <arg name="square"/>
        <node pkg="camera_calibration" type="cameracalibrator.py"
            name="calibrator" output="screen"
            args="-p $(arg pattern) -s $(arg size) -q $(arg square)
            image:=/$(arg camera)/image_raw camera:=/$(arg camera)">
        </node>
    </group>
</launch>
